client/app/components/application.coffee

```diff
@@ -40,153 +30,58 @@ MessageActionCreator = require '../actions/message_action_creator'
     (based on:
         https://medium.com/react-tutorials/react-backbone-router-c00be0cf1592)
 ###
-module.exports = Application = React.createClass
+
+Application = React.createClass
     displayName: 'Application'
 
     mixins: [
-        StoreWatchMixin Stores
-        RouterMixin
         TooltipRefesherMixin
     ]
 
-    render: ->
-        # Shortcut
-        # TODO: Improve the way we display a loader when app isn't ready
-        layout = @props.router.current
-        return div null, t "app loading" unless layout?
-
-        disposition = LayoutStore.getDisposition()
-        isCompact   = LayoutStore.getListModeCompact()
-        fullscreen  = LayoutStore.isPreviewFullscreen()
-        previewSize = LayoutStore.getPreviewSize()
+    getDefaultProps: ->
+        ApplicationGetters.getProps 'application'
```

Comment:
@aenario :
`ActionCreator` are the `controller` of all the `states` of the Application also known as `Stores`.
After dispatching an `Action` we can dispatch a `success` or `error` action to update application (ie. fetching message) and :
 - make redirections (and remove it from `views`),
 - update data (from this redirection if needed).
 
Handling that kind of action should prevent us to `listenTo` `Stores`; that's why I think we can remove all `StoreWatch` logic.

What do you think?